src/lib/voiceSearchHelper/__tests__/index-test.js

```diff
@@ -0,0 +1,154 @@
+import VoiceSearchHelper from '../';
+
+const getHelper = opts =>
+  new VoiceSearchHelper(
+    opts || {
+      onQueryChange: () => {},
+      onStateChange: () => {},
+    }
+  );
+
+describe('VoiceSearchHelper', () => {
+  beforeEach(() => {
+    delete window.webkitSpeechRecognition;
+    delete window.SpeechRecognition;
+  });
+
+  it('test nothing', () => {
+    expect(true).toBeTruthy();
+  });
+
+  it('has initial state correctly', () => {
+    const helper = getHelper();
+    expect(helper.getState()).toMatchSnapshot();
+  });
+
+  it('is not supported', () => {
+    const helper = getHelper();
+    expect(helper.isSupportedBrowser()).toBeFalsy();
+  });
+
+  it('is not listening', () => {
+    const helper = getHelper();
+    expect(helper.isListening()).toBeFalsy();
+  });
+
+  it('is supported with webkitSpeechRecognition', () => {
+    window.webkitSpeechRecognition = () => {};
+    const helper = getHelper();
+    expect(helper.isSupportedBrowser()).toBeTruthy();
+  });
+
+  it('is supported with SpeechRecognition', () => {
+    window.SpeechRecognition = () => {};
+    const helper = getHelper();
+    expect(helper.isSupportedBrowser()).toBeTruthy();
+  });
+
+  it('works with mock SpeechRecognition (searchAsYouSpeak:false)', () => {
+    let that;
+    window.SpeechRecognition = jest.fn().mockImplementation(() => ({
+      start() {
+        that = this;
+      },
+    }));
+    const onQueryChange = jest.fn();
+    const onStateChange = jest.fn();
+    const helper = getHelper({
+      onQueryChange,
+      onStateChange,
+    });
+    const { getState } = helper;
+    helper.toggleListening({ searchAsYouSpeak: false });
+    expect(onStateChange).toHaveBeenCalled();
+    expect(getState().status).toEqual('askingPermission');
+    that.onstart();
+    expect(getState().status).toEqual('waiting');
+    that.onresult({
+      results: [
+        (() => {
+          const obj = [
+            {
+              transcript: 'Hello World',
+            },
+          ];
+          obj.isFinal = true;
+          return obj;
+        })(),
+      ],
+    });
+    expect(getState().status).toEqual('recognizing');
+    expect(getState().transcript).toEqual('Hello World');
+    expect(getState().isSpeechFinal).toBeTruthy();
+    expect(onQueryChange).not.toHaveBeenCalled();
```

Comment:
`toHaveBeenCalledTimes(0)` usually gives better error messages and is easier to update in the long run.